File: Leetcode 0228 Summary Ranges.go
// 20250806
func summaryRanges(nums []int) []string {
    var ans []string
    i := 0
    start := 0
    for i < len(nums){
        start = i
        for i + 1 < len(nums) && nums[i+1] == nums[i] + 1{
            i+=1
        }
        if i == start{
            ans = append(ans, strconv.Itoa(nums[start]))
        } else{
            ans = append(ans, strconv.Itoa(nums[start])+"->"+strconv.Itoa(nums[i]))
        }
        i+=1
    }
    return ans
}

File: Leetcode 0228 Summary Ranges.py
# 20250806
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        elif len(nums) == 1:
            return [f"{nums[0]}"]

        ans = []
        i = 0
        while i < len(nums):
            start = i
            while i + 1 <len(nums) and nums[i+1]-nums[i] == 1:
                i+=1
            if start == i:
                ans.append(f"{nums[i]}")
            else:
                ans.append(f"{nums[start]}->{nums[i]}")
            i += 1
        return ans
